extensions/wikia/Flags/FlagsHelper.class.php

```diff
@@ -143,11 +144,39 @@ public function shouldDisplayFlags() {
 			/* Don't display flags on edit pages that are content namespaces */
 			&& !in_array(
 				$wgRequest->getVal( 'action', 'view' ),
-				[ 'edit', 'formedit' , 'history' ]
+				[ 'edit', 'formedit' , 'history', 'visualeditor' ]
 			);
 	}
 
 	/**
+	 * Checks if flags should be displayed on a page
+	 * @return bool
+	 */
+	public function shouldDisplayFlags() {
+		global $wgTitle;
+
+		return
+			/* Don't display flags for non existent pages */
+			$wgTitle->exists()
+			/* Display flags only on content namespaces */
+			&& \Wikia::isContentNamespace();
+	}
+
+	/**
+	 * Checks if flags can be edited on current page to decide whether include edit modal
+	 * @return bool
+	 */
+	public function areFlagsEditable() {
+		global $wgUser;
+		return
+			/* Check condition for view */
+			$this->shouldDisplayFlags()
+			/* Don't display flags when user is not allowed to edit */
+			&& $wgUser instanceof \User
+			&& $wgUser->isAllowed( 'edit' );
```

Comment:
I'd consider using `Title::userCan` instead of `User::isAllowed` as `isAllowed` just checks if the user has the edit right, not if they can edit the current page. `Title::userCan` checks if they can edit the current page including checking if it is protected and whether the user has the rights to edit pages at that protection level.